src/vellum/workflows/events/types.py

```diff
@@ -42,9 +38,21 @@ def default_serializer(obj: Any) -> Any:
     )
 
 
+class CodeResourceDefinition(UniversalBaseModel):
+    name: str
+    module: List[str]
+
+
+VellumCodeResourceDefinition = Annotated[
+    CodeResourceDefinition,
+    BeforeValidator(lambda d: (d if type(d) is dict else serialize_type_encoder(d))),
```

Comment:
sets up __class__ to be encoded before trying to cast to CRD type

Reply:
[q] Do we need this bc the dict being fed into pydantic could contain a `CodeResourceDefinition` instead of a `dict`? when does this happen?

Reply:
This happens before validation! We could pass in an argument of both a dictionary hat's being passed in when instantiating via something like NodeParentContext( ** previous_parent_context ) and when creating a new context like NodeParentContext(node_definition=node.__class__ ... ) so we technically could also support someone directly using a "CodeResourceDefinition", but they'd still need to transform that __class__ . This happens before validation tho, so it'll still run through the validator.